### PR TITLE
2056 additional deepgrow workflow events

### LINKED ISSUE #2056: additional events in the deepgrow interaction engine
**Is your feature request related to a problem? Please describe.**
This is a feature request for adding extra engine events within the click simulation loops during the deepgrow model training:
https://github.com/Project-MONAI/MONAI/blob/abad8416153e67aac04417bbd9398f334b9c0912/monai/apps/deepgrow/interaction.py#L61-L77

the main benefit is to have flexible simulation handlers attached to the inner loops

cc @danieltudosiu @diazandr3s @SachidanandAlle

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

Fixes #2056

### Description
this PR adds two new events to IterationEvents, which are useful for advanced workflows and debugging.
 it's tested with the deepgrow workflow

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/deepgrow/interaction.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ def __init__(
         self.key_probability = key_probability
 
     def attach(self, engine: Union[SupervisedTrainer, SupervisedEvaluator]) -> None:
-        if not engine.has_event_handler(self, Events.ITERATION_STARTED):
-            engine.add_event_handler(Events.ITERATION_STARTED, self)
+        if not engine.has_event_handler(self, IterationEvents.ITERATION_STARTED):
```
Comment (Nic-Ma) [inline monai/apps/deepgrow/interaction.py:54]: I think `IterationEvents` doesn't have `ITERATION_STARTED` event, no need to change this line.
Thanks.

**Review by Nic-Ma — COMMENTED** [review]
Thanks for your quick PR.
I put some comments for a minor issue, others look good to me.

Thanks.

**Comment by wyli** [comment]
the`attach` method is in fact incorrect, `interaction.attach(engine)` gives
```
ValueError: Error adding <monai.apps.deepgrow.interaction.Interaction object at 0x7fa719b25d10> 
'handler': takes parameters ['engine', 'batchdata'] but will be called with [](missing a required argument: 'engine').
```

**Review by Nic-Ma — APPROVED** [review]
Thanks for your quick update.
Looks good to me now.

Thanks.

**Pull request merged**